pkg/command/root.go

```diff
@@ -144,8 +201,11 @@ func setupTargetPlugins() error {
 
 	// Insert the plugin commands under the appropriate target command
 	for i := range installedPlugins {
-		if cmd, exists := mapTargetToCmd[installedPlugins[i].Target]; exists {
-			cmd.AddCommand(cli.GetCmdForPlugin(&installedPlugins[i]))
+		if targetCmd, exists := mapTargetToCmd[installedPlugins[i].Target]; exists {
+			cmd := cli.GetUnmappedCmdForPlugin(&installedPlugins[i])
+			if cmd != nil {
+				targetCmd.AddCommand(cmd)
+			}
```

Comment:
Right below this line we look if a target should be hidden because it has no plugins.
This logic needs to be moved until the mapped plugins have been installed.

You can try it now by removing all plugins and then installing a single mapped plugin under the kubernetes target.  You will notice that the `kubernetes` target is hidden from the help and shell completion, even though the mapped plugin is invokable under that target.

Reply:
Also, this function uses `pluginsupplier.GetInstalledPlugins()` and therefore does not respect the new `SupportedContextType` field.  This means that if a plugin specifies `SupportedContextType: "tanzu"` without an `InvokedAs` remap, the CLI will remove that plugin if there is no "tanzu" context active, BUT it will not remove the plugin from under targets.

Reply:
Nice catch! Updated to filter plugins via ActiveContext as well, and added unit test case